data/tutorials/getting-started/3_01_ocaml_on_windows.md

```diff
@@ -6,8 +6,211 @@ description: >
+different shell? [y/n] n
+```
+
+One detail to address about having a fully functional opam installation is
+related to Git.
+
+On Windows, there are many ways to have a functioning Git installation. Opam will
+look for a compatible Git and, if none is found, it will prompt you with a set of
+options to install one. 
+
+```shell-session
+<><> Git ><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>  🐫
+Cygwin Git is functional but can have credentials issues for private repositories, we recommend using:
+  - Install via 'winget install Git.Git'
+  - Git for Windows can be downloaded and installed from https://gitforwindows.org
+
+Which Git should opam use?
+> 1. Install Git with along with Cygwin internally
+  2. Enter the location of your Git installation
+  3. Abort initialisation to restart your shell.
+
+[1/2/3]
+```
+
+Opam's default behaviour when initialising is to install a fresh `switch` as
+well as an ocaml compiler of version `> 4.05`.
+
+```shell-session
+<><> Creating initial switch 'default' (invariant ["ocaml" {>= "4.05.0"}] - initially with ocaml-base-compiler)
+
+<><> Installing new switch packages <><><><><><><><><><><><><><><><><><><><>  🐫
+Switch invariant: ["ocaml" {>= "4.05.0"}]
+
+The following system packages will first need to be installed:
+    mingw64-x86_64-gcc-core
+
+<><> Handling external dependencies <><><><><><><><><><><><><><><><><><><><>  🐫
+
+[Installing external dependencies]
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><>  🐫
+∗ installed arch-x86_64.1
+∗ installed base-bigarray.base
+∗ installed base-threads.base
+∗ installed base-unix.base
+∗ installed host-arch-x86_64.1
+∗ installed host-system-mingw.1
+⬇ retrieved mingw-w64-shims.0.2.0  (https://opam.ocaml.org/cache)
+⬇ retrieved flexdll.0.43  (https://opam.ocaml.org/cache)
+∗ installed flexdll.0.43
+∗ installed ocaml-options-vanilla.1
+∗ installed conf-mingw-w64-gcc-x86_64.1
+∗ installed ocaml-env-mingw64.1
+∗ installed system-mingw.1
+⬇ retrieved ocaml-config.3  (2 extra sources)
+⬇ retrieved ocaml-config.3  (2 extra sources)
+∗ installed mingw-w64-shims.0.2.0
+⬇ retrieved ocaml-base-compiler.5.2.0  (https://opam.ocaml.org/cache)
+∗ installed ocaml-base-compiler.5.2.0
+∗ installed ocaml-config.3
+∗ installed ocaml.5.2.0
+∗ installed base-domains.base
+∗ installed base-nnp.base
+Done.
+# Run for /f "tokens=*" %i in ('opam env --switch=default') do @%i to update the current shell environment
+```
+
+You can see that in this specific run of the command, opam chose `mingw` as
+a C compiler but know that you can very well choose an alternative to it
+instead, like `msvc` with the following command:
+
+```shell-session
+$ opam install system-msvc
+```
+
+Once your environment has been updated with the given hint, you will have a
+fully functional OCaml compiler available to you:
+
+```shell-session
+> for /f "tokens=*" %i in ('opam env --switch=default') do @%i
```

Comment:
Correct me if  I'm wrong, but if you install winget in a terminal, and then install opam using winget, any subsequent opam like `opam env` will fail with Not Found, right? Because opam won't be in the PATH.  If so, I think we need to insert a step to close and reopen the terminal.